modules/homa-lite/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+		/// staking_total_issuance, liquid_total_issuance]
+		BatchProcessed(EraIndex, Balance, Balance),
+
+		/// The user has claimed some Liquid Currency. [batch, user, amount]
+		LiquidCurrencyClaimed(EraIndex, T::AccountId, Balance),
+
+		/// The relay-chain's stash account ID has been updated.
+		RelaychainStashAccountUpdated(T::AccountId),
+	}
+
+	/// Stores the amount of Staking currency the user has exchanged.
+	/// PendingAmount: double_map: (batch: EraIndex, user: T::AccountId) -> amount: Balance
+	#[pallet::storage]
+	#[pallet::getter(fn pending_amount)]
+	pub type PendingAmount<T: Config> =
+		StorageDoubleMap<_, Blake2_128Concat, EraIndex, Blake2_128Concat, T::AccountId, Balance, ValueQuery>;
+
+	/// The total issuance info for each batch. Used to calculate Staking to Liquid exchange rate.
+	/// BatchTotalIssuanceInfo: map: batch: EraIndex -> batch_total: TotalIssuanceInfo
+	#[pallet::storage]
+	#[pallet::getter(fn batch_total_issuance_info)]
+	pub type BatchTotalIssuanceInfo<T: Config> =
+		StorageMap<_, Blake2_128Concat, EraIndex, TotalIssuanceInfo, OptionQuery>;
+
+	/// The batch that is currency active
+	/// CurrentBatch: value: batch: EraIndex
+	#[pallet::storage]
+	#[pallet::getter(fn current_batch)]
+	pub type CurrentBatch<T: Config> = StorageValue<_, EraIndex, ValueQuery>;
+
+	/// The account in which the staking currency goes into to be transferred to the Relay chain.
+	/// RelaychainStashAccount: value: stash_account: AccountId
+	#[pallet::storage]
+	#[pallet::getter(fn relaychain_stash_account)]
+	pub type RelaychainStashAccount<T: Config> = StorageValue<_, T::AccountId, OptionQuery>;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Request to mint some Liquid currency, by locking up the given amount of Staking
+		/// currency. The exchange does not happen immediately, but on when the batch is processed
+		/// The user then needs to manually claim the Liquid currency once it is ready.
+		///
+		/// Parameters:
+		/// - `amount`: The amount of Staking currency to be exchanged.
+		//#[pallet::weight(< T as Config >::WeightInfo::request_mint())]
+		#[pallet::weight(0)]
+		#[transactional]
+		pub fn request_mint(origin: OriginFor<T>, amount: Balance) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let current_batch = Self::current_batch();
+			let staking_currency_id = T::StakingCurrencyId::get();
+
+			let stash_account = &Self::relaychain_stash_account().ok_or(Error::<T>::RelaychainStashAccountNotSet)?;
+
+			// TODO: Cross-chain transfer to the Relaychain via XCM
+			T::Currency::transfer(staking_currency_id, &who, &stash_account.clone(), amount)?;
+
+			let current = PendingAmount::<T>::get(current_batch, &who);
+			// Due to the math on Total Issuance, it is literally impossible to overflow here.
+			let new_value = current + amount;
+			PendingAmount::<T>::insert(current_batch, &who, new_value);
```

Comment:
Use `mutate` instead of get and insert. And use `.checked_add` with `.expect("...proof of no overflow")`